streams/runners/storm/runtime/src/test/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroKafkaSpoutTranslatorTest.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2017 Hortonworks.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *   http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package com.hortonworks.streamline.streams.runtime.storm.spout;
+
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+import static org.junit.Assert.assertEquals;
+
+public class AvroKafkaSpoutTranslatorTest {
+
+    @Test
+    public void testByteBufferInputStream() throws Exception {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(1000);
+        AvroKafkaSpoutTranslator.ByteBufferInputStream is = new AvroKafkaSpoutTranslator.ByteBufferInputStream(byteBuffer);
+        byte[] dest = new byte[1000];
+        int res = is.read(dest, 0, 1000);
+        assertEquals(1000, res);
+    }
+
+    @Test
+    public void testByteBufferInputStreamNonZeroOffset1() throws Exception {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(1000);
+        AvroKafkaSpoutTranslator.ByteBufferInputStream is = new AvroKafkaSpoutTranslator.ByteBufferInputStream(byteBuffer);
+        byte[] dest = new byte[1000];
+        int res = is.read(dest, 100, 900);
+        assertEquals(900, res);
+    }
+
+    @Test
+    public void testByteBufferInputStreamNonZeroOffset2() throws Exception {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(1000);
```

Comment:
A method can be extracted which takes offset and len, which can be used in the two tests above.